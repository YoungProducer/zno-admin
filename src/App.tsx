import React, { useState, useEffect } from 'react';

import Routes from 'containers/routes';
import Notifier from 'components/Notifier';

interface IApp {
    loggedIn: boolean;
    emailAfterSignUp: string;
    fetchMe: Function;
}

const App = ({
    loggedIn,
    emailAfterSignUp,
    fetchMe,
}: IApp) => {

    return(
        <>
            <Notifier />
            <Routes />
        </>
    );
};

export default App;
