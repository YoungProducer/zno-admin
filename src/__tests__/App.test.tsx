/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 15 February 2020
 *
 * Create test suites for App component.
 */

// External imports
import React from 'react';
import { shallow } from 'enzyme';

// Application's imports
import App from '../App';

describe('App component', () => {
    const requiredProps = {
        loggedIn: false,
        emailAfterSignUp: '',
        fetchMe: jest.fn(),
    };

    test('Is matches snapshot', () => {
        const root = shallow(<App {...requiredProps}/>);

        expect(root).toMatchSnapshot();
    });
});
