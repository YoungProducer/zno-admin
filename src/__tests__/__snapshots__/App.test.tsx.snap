// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Is matches snapshot 1`] = `
<Fragment>
  <Connect(Component) />
  <Connect(Component) />
</Fragment>
`;
